cli/index.ts

```diff
@@ -0,0 +1,126 @@
+import type { Args } from "https://deno.land/std@0.200.0/flags/mod.ts";
+import { parse } from "https://deno.land/std@0.200.0/flags/mod.ts";
+import { open } from "https://deno.land/x/open@v0.0.6/index.ts";
+
+export function getLink(
+  space: string,
+  filters: Record<string, string>,
+  columns: string[],
+): string {
+  const kibanaFilters = Object.entries(filters).map(([key, value]) => {
+    return `(query:(match_phrase:(${key}:'${value}')))`;
+  });
+
+  // The `#/` at the end is important for Kibana to correctly parse the query string
+  // The `URL` object moves this to the end of the string, which breaks the link.
+  const base = `https://logs.gutools.co.uk/s/${space}/app/discover#/`;
+
+  const query = {
+    ...(kibanaFilters.length > 0 && {
+      _g: `(filters:!(${kibanaFilters.join(",")}))`,
+    }),
+    ...(columns.length > 0 && {
+      _a: `(columns:!(${columns.join(",")}))`,
+    }),
+  };
+
+  const queryString = Object.entries(query)
+    .map(([key, value]) => `${key}=${value}`)
+    .join("&");
+
+  return `${base}?${queryString}`;
+}
+
+function parseArguments(args: string[]): Args {
+  return parse(args, {
+    boolean: ["follow"],
+    negatable: ["follow"],
+    string: ["space", "stack", "stage", "app"],
+    collect: ["column", "filter"],
+    stopEarly: false,
+    "--": true,
+    default: {
+      follow: true,
+      column: ["message"],
+      space: "default",
+      filter: [],
+    },
+  });
+}
+
+function escapeColon(str: string): string {
+  return str.includes(":") ? `'${str}'` : str;
+}
+
+function parseFilters(filter: string[]): Record<string, string> {
+  return filter.reduce((acc, curr) => {
+    const [key, value] = curr.split("=");
+    return { ...acc, [escapeColon(key)]: value };
```

Comment:
Can maybe add some logic here to only add a key value pair if the value is not undefined?
That way you can get rid of `removeUndefined`

Reply:
I'd prefer to keep to separate, single responsibility, functions as they are easier to test. That said, the call to `escapeColon` isn't making `parseFilters` single responsibility, nor are there tests! I'll refactor, and add the tests.

Reply:
Tests added in 40e18140851eb83d9f15d79000ad435addb9d96e.